aquacomputer_d5next.c

```diff
@@ -390,11 +430,25 @@ static int aqc_read(struct device *dev, enum hwmon_sensor_types type, u32 attr,
 		break;
 	case hwmon_pwm:
 		if (priv->fan_ctrl_offsets != NULL) {
-			ret = aqc_get_ctrl_val(priv, priv->fan_ctrl_offsets[channel]);
-			if (ret < 0)
-				return ret;
+			switch (attr) {
+			case hwmon_pwm_enable:
+				ret = aqc_get_u8_val(priv, priv->fan_ctrl_offsets[channel] - 1);
```

Comment:
this byte isn't guaranteed to be in the same spot every time (even though it will be...)

I think the code would be cleaner with it as it's own little table of offsets.

Reply:
_or_ perhaps we could use a `define` for the `-1` to make it more obvious what we're doing. Up to you, I suppose?

Reply:
ooh elegant, good idea!